internal/vault/utils.go

```diff
@@ -3,9 +3,51 @@
 
 package vault
 
-import "strings"
+import (
+	"context"
+	"os"
+	"strings"
+
+	"github.com/hashicorp/vault-secrets-operator/internal/common"
+	appsv1 "k8s.io/api/apps/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	operatorDeploymentDefaultName = "vault-secrets-operator-controller-manager"
+	operatorDeploymentKind        = "Deployment"
+	operatorDeploymentAPIVersion  = "apps/v1"
+)
 
 // JoinPath for Vault requests.
 func JoinPath(parts ...string) string {
 	return strings.Join(parts, "/")
 }
+
+func getOperatorDeployment(c client.Client) (*appsv1.Deployment, error) {
+	operatorDeploymentName := ""
+	if operatorDeploymentName = os.Getenv("OPERATOR_DEPLOYMENT_NAME"); operatorDeploymentName == "" {
```

Comment:
I wonder if there might be a better way of getting the deployment name. This approach feels a bit brittle. Maybe we could find the Deployment based on some known label?